### DISCUSSION TITLE
How do I effectively insert code into a JS file?

### DISCUSSION THREAD

**Discussion by just-a-hriday** [ScratchAddons/ScratchAddons#3493] [Help]
So, I was going around trying to create my first addon - but then I realized that I don't know how browser extensions work.

I think that all I need to do is to insert a bit of code into a certain JS function in scratch-gui, but I have no idea how to do that externally from a browser extension.

Does someone else know how to achieve that?

**Comment by apple502j** [comment]
Can you post exactly what you want to do - which file, which function, and what code?

**Reply by just-a-hriday** [reply]
I want to add a loop that takes all the costumes from sprites in the backpack and puts them into the currently selected sprite - basically, I want dragging a sprite into the costume panel result in copying all the sprite's costumes into the selected sprite.
I believe this can be done by adding exactly that to the `handleDrop` function in [scratch-gui/src/containers/target-pane.jsx](https://github.com/LLK/scratch-gui/blob/a2766c36e448699e1811415c6114c7a015ebee0f/src/containers/target-pane.jsx#L196).

I think I know exactly what to add, but there's no way to add it to scratch-gui itself and then test it, so I decided to make an addon that did that - which is what brought me here.

**Reply by FunctionalMetatable** [reply]
> 
> 
> I want to add a loop that takes all the costumes from sprites in the backpack and puts them into the currently selected sprite - basically, I want dragging a sprite into the costume panel result in copying all the sprite's costumes into the selected sprite.
> I believe this can be done by adding exactly that to the `handleDrop` function in [scratch-gui/src/containers/target-pane.jsx](https://github.com/LLK/scratch-gui/blob/a2766c36e448699e1811415c6114c7a015ebee0f/src/containers/target-pane.jsx#L196).
> 
> I think I know exactly what to add, but there's no way to add it to scratch-gui itself and then test it, so I decided to make an addon that did that - which is what brought me here.

I believe you can directly interface with react components/containers from the dom

**Reply by just-a-hriday** [reply]
> I believe you can directly interface with react components/containers from the dom

Umm... Uhh... I don't know _that_ much JS.

**Reply by apple502j** [reply]
Maybe check folders addon? (Kinda complicated tho)

**Reply by WorldLanguages** [reply]
Editor addons are complicated.

**Reply by retronbv** [reply]
> Editor addons are complicated.

*Addons* are complicated

**Reply by cobaltt7** [reply]
Not necessarily
